### ISSUE TITLE
EarlyStopping restore_best_weights argument similar to Keras

### ISSUE THREAD

**Issue by vedal** [PyTorchLightning/pytorch-lightning#1415]
## 🚀 Feature
EarlyStopping argument `restore_best_weights` restores model weights from the epoch with the best value of the monitored loss, similar to Keras. Would not need to run ModelCheckpoint every epoch.

### Motivation
Good to have to avoid even more pytorch boilerplate

### Alternatives
Calling `ModelCheckpoint` ever epoch

**Comment by philip30** [comment]
this is similar to my issue #1395

**Comment by Borda** [comment]
@vedal let's keep just one - #1395
